src/sql/src/plan/statement/ddl.rs

```diff
@@ -1433,6 +1433,30 @@ fn kafka_sink_builder(
     };
     let config_options = kafka_util::extract_config(with_options)?;
 
+    let avro_key_fullname = match with_options.remove("avro_key_fullname") {
+        Some(Value::String(s)) => Some(s),
+        None => None,
+        Some(_) => bail!("avro_key_fullname must be a string"),
+    };
+
+    if key_desc_and_indices.is_none() && avro_key_fullname.is_some() {
+        bail!("Cannot specify avro_key_fullname without a corresponding KEY field");
+    }
+
+    let avro_value_fullname = match with_options.remove("avro_value_fullname") {
+        Some(Value::String(s)) => Some(s),
+        None => None,
+        Some(_) => bail!("avro_value_fullname must be a string"),
+    };
+
+    if key_desc_and_indices.is_some() {
+        match (avro_key_fullname.is_some(), avro_value_fullname.is_some()) {
+            (true, false) | (false, true) =>
```

Comment:
this whole bit is an exclusive or, wasn't sure what the most idiomatic way to write it would be. I've never found `^` to be super readable, but that might just be from unfamiliarity

Reply:
I would have written `!(a ^ b)` personally but either is fine IMO, as is `(a && b) || !(a || b)`.